src/lightning/data/streaming/combined.py

```diff
@@ -43,53 +49,120 @@ def __init__(
             self._weights = [w / sum(weights) for w in weights]
 
         self._iterator: Optional[_CombinedDatasetIterator] = None
+        self._use_streaming_dataloader = False
+        self._num_samples_yielded = None
+        self._current_epoch = 0
+
+    def set_epoch(self, current_epoch: int) -> None:
+        self._current_epoch = current_epoch
+        for dataset in self._datasets:
+            dataset.set_epoch(current_epoch)
```

Comment:
Let's add a docstring here, and in which case the user is required to call this and where in the loop (beginning of epoch?). I assume this is for shuffling reasons? If you wouldn't call it, would it only affect shuffling?

Reply:
done !